test/meta_testing_tools.jl

```diff
@@ -0,0 +1,71 @@
+# This is tools for testing ChainRulesTestUtils itself
+# if they were less nasty in implementation we might consider moving them to a package
+# MetaTesting.jl
+
+"""
+    metatest_get_failures(f)
+
+`f` should be a function that takes no argument, and calls some code that used `@test`.
+Invoking it via `metatest_get_failures(f)` will prevent those `@test` being added to the
+current testset, and will return a collection of all nonpassing test results.
+"""
+function metatest_get_failures(f)
```

Comment:
Should this just be called
```suggestion
function get_failures(f)
```
?

Reply:
yeah, `extract_failures` or `fail_results` (since they are `Test.Fail <: Test.Result`) or just  `failures`?

Reply:
They are any non-pass.
So they could be Errors, or some other cause, I don't recall what `@test_broken` returns.
Arguably we should call it `nonpasses`

Reply:
yeah `nonpasses` is better (to avoid the `failures` = `Fail` confusion i had)

`test_broken` also treats all non-passes the same
```julia
julia> @test_broken 1 == 2
Test Broken
  Expression: 1 == 2

julia> @test_broken error()
Test Broken
  Expression: error()
```